engine/database/transaction.go

```diff
@@ -1,8 +1,33 @@
 package database
 
 // Transaction represents a transaction in the database.
-// Fixme: This is not integrated with out db engine yet.
 type Transaction struct {
 	ID       string `gorm:"type:char(64);primaryKey"`
 	TxStatus TxStatus
+
+	Outputs []*Output `gorm:"foreignKey:TxID"`
+	Data    []*Data   `gorm:"foreignKey:TxID"`
+	Inputs  []*Output `gorm:"foreignKey:SpendingTX"`
+}
+
+// TableName implements gorm.Tabler to override automatic table naming.
+// NOTE: This is because we have already a legacy table named "transactions".
+// TODO: Remove this when we have migrated all data.
+func (t *Transaction) TableName() string {
+	return "new_transactions"
```

Comment:
### minor
Maybe `tracked_transactions` so even if we forgot to rename and need another new transactions then we won't have new_new_transactions 🙈

Reply:
Renamed to `TranckedTransaction`👍So the custom "namer" is no longer necessary.